fastly/resource_fastly_service_v1_dynamicsnippet_test.go

```diff
@@ -0,0 +1,222 @@
+package fastly
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	gofastly "github.com/fastly/go-fastly/fastly"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestResourceFastlyFlattenDynamicSnippets(t *testing.T) {
+
+	cases := []struct {
+		remote []*gofastly.Snippet
+		local  []map[string]interface{}
+	}{
+		{
+			remote: []*gofastly.Snippet{
+				{
+					Name:     "recv_test_01",
+					Type:     gofastly.SnippetTypeRecv,
+					Priority: 110,
+					Content:  "if ( req.url ) {\n set req.http.my-snippet-test-header = \"true\";\n}",
+					Dynamic:  1,
+				},
+				{
+					Name:     "recv_test_02",
+					Type:     gofastly.SnippetTypeRecv,
+					Priority: 110,
+					Content:  "if ( req.url ) {\n set req.http.my-snippet-test-header = \"true\";\n}",
+					Dynamic:  0,
+				},
```

Comment:
The second object of the remote field will never be compared against a local object! In order to test that the `flattenDynamicSnippets` function ignores `static` snippets the test should look like:

```
func TestResourceFastlyFlattenSnippets(t *testing.T) {

	cases := []struct {
		remote []*gofastly.Snippet
		local  []map[string]interface{}
	}{
		{
			remote: []*gofastly.Snippet{
				{
					Name:     "recv_test",
					Type:     gofastly.SnippetTypeRecv,
					Priority: 110,
					Content:  "if ( req.url ) {\n set req.http.my-snippet-test-header = \"true\";\n}",
				},
			},
			local: []map[string]interface{}{
				{
					"name":     "recv_test",
					"type":     gofastly.SnippetTypeRecv,
					"priority": 110,
					"content":  "if ( req.url ) {\n set req.http.my-snippet-test-header = \"true\";\n}",
				},
			},
		},
		{
			remote: []*gofastly.Snippet{
				{
					Name:     "recv_test",
					Type:     gofastly.SnippetTypeRecv,
					Priority: 110,
					Content:  "if ( req.url ) {\n set req.http.my-snippet-test-header = \"true\";\n}",
					Dynamic:  1,
				},
			},
			local: []map[string]interface{}(nil),
		},
	}

	for _, c := range cases {
		out := flattenSnippets(c.remote)
		if !reflect.DeepEqual(out, c.local) {
			t.Fatalf("Error matching:\nexpected: %#v\n got: %#v", c.local, out)
		}
	}

}
```